django_phaxio/views.py

```diff
@@ -0,0 +1,80 @@
+import hashlib
+import hmac
+import json
+import logging
+
+from django.core.exceptions import PermissionDenied
+from django.utils.decorators import method_decorator
+from django.views.decorators.csrf import csrf_exempt
+from django.views.generic import View
+from django.http.response import JsonResponse
+
+from . import DIRECTION
+from .conf import settings
+from .signals import phaxio_callback
+
+logger = logging.getLogger(__name__)
+
+DIRECTION_MAP = {
+    'send': DIRECTION.send,
+    'received': DIRECTION.received,
+}
+
+
+class PhaxioCallbackView(View):
+    """A view to receive WebHooks from Phaxio."""
+
+    @method_decorator(csrf_exempt)
+    def dispatch(self, request, *args, **kwargs):
+        """Disable CSRF but enable token based security."""
+        self.validate_signature(request)
+        return super(PhaxioCallbackView, self).dispatch(request, *args, **kwargs)
+
+    def post(self, request):
+        """Process WebHook and send :func:`.phaxio_callback` signal."""
+        fax = request.POST["fax"]
+        direction = request.POST['direction']
+        is_test = request.POST['is_test']
+        is_test = is_test in [1, '1', True, 'true', 'True']
```

Comment:
@syphar DRF is too big for this. Plus this a form/multipart request not a json request.